providers/kit/metadata.go

```diff
@@ -0,0 +1,90 @@
+package kit
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+var (
+	perPage          = "per_page" // nolint:gochecknoglobals
+	metadataPageSize = "1"        // nolint:gochecknoglobals
+)
+
+// ListObjectMetadata creates metadata of object via reading objects using Kit API.
+func (c *Connector) ListObjectMetadata(ctx context.Context,
```

Comment:
This shouldn't belong to the Read PR.

Reply:
I created a branch from the metadata branch that's why these changes are showing. As metadata functionality is not merged in the main I have done this way because I need some functionality added for metadata